collectors/python.d.plugin/dockerd/README.md

```diff
@@ -5,6 +5,12 @@ Module monitor docker health metrics.
 **Requirement:**
 
 -   `docker` package, required version 3.2.0+
+- `netdata` user should be added to the group `docker` so it can access the socket. Read more at the [docker docks](https://docs.docker.com/install/linux/linux-postinstall/).
+```bash
+sudo groupadd docker #create docker group
+sudo usermod -aG docker netdata #add user
```

Comment:
i think `docker` group is already exists if `docker` is installed on the system.
And `netdata` is added to the group during install

https://github.com/netdata/netdata/blob/1094175c3bd08ecf5c83c91605e9fd1d02052108/netdata-installer.sh#L542-L558

Reply:
But you are right, group requirement should be reflected in the readme

Reply:
In my setup (ubuntu 16) netdata user did not belong to the docker group and thus the plugin did not work out-of-the-box. It got activated after I added the user to the group.

Reply:
Netdata is added to those groups during install and update if the groups exist

```
 --- Creating standard user and groups for netdata ---
 --- Adding group 'netdata' ---
Group 'netdata' already exists.
 --- Adding user 'netdata' ---
User 'netdata' already exists.
 --- Assign user 'netdata' to required groups ---
User 'netdata' is already in group 'docker'.
Group 'nginx' does not exist.
Group 'varnish' does not exist.
Group 'haproxy' does not exist.
User 'netdata' is already in group 'adm'.
Group 'nsd' does not exist.
User 'netdata' is already in group 'proxy'.
Group 'squid' does not exist.
Group 'ceph' does not exist.
Group 'nobody' does not exist.
```

> netdata-installer.sh
> netdata-updater.sh

Reply:
My main complain here is creating the group

> sudo groupadd docker #create docker group

User dont need to create any groups manually

Reply:
I am referring to this one: https://github.com/netdata/netdata/issues/7328 
I experienced the same thing with dockerd. Maybe depending on the system, the installer fails occasionally to add the groups?

Reply:
The problem in #7328 could be order:
 - install netdata
 - install varnish

@OdysLam 

you can run 

> [netdata-installer.sh](https://github.com/netdata/netdata/blob/0f72fd029099080c9080eedfa6493aa1a5ac7ab9/netdata-installer.sh#L542-L558)

and check output `Creating standard user and groups for netdata` part. It is a bash script and it the only req is `if [ "${UID}" -eq 0 ]; then`. Experiment with it if you have time.

Reply:
I will experiment with it.

But the notion still stands: even if the problem is the result of first installing netdata and then varnish, shouldn't the user be informed of the requirement, so in case it doesn't work, he can troubleshoot it easily without searching github issues for an awswer.

How should the user be protected from that common error (first install netdata and then varnish) ?

Reply:
I agree that we need to add "being in a group" requirement to the readme. In addition we could check if user in a group and warn if it is not or smth like that

```python
import os, grp
[grp.getgrgid(g).gr_name for g in os.getgroups()]
```

Reply:
Would it be possible to warn the user _without using the debug feature_? Maybe as a text replacement of the chart or as an alert?

What I have in mind, is that ideally, I would want Netdata to alert me of that without making me search for it (either Github, python debug or ask around).

Reply:
No, the flow is to look in the error.log/debug mode. We need to add `troubleshooting` section to very readme.

> Maybe as a text replacement of the chart or as an alert?

There is no chart if it is doesnt work

Reply:
ok,but I think that the user should be able to troubleshoot more easily, ideally directly from the interface.

I will add the troubleshoot sections in both readmes.

Reply:
Impossible, error.log/debug is the best and **well-known approach**. Usually there is a lot of details in the debug, there is no way to display them in gui.

Reply:
so as a PR, we keep the mention in the docs and PR a small addition in the code that checks for the user in the group.